bitcoin/core/__init__.py

```diff
@@ -422,11 +461,9 @@ def GetTxid(self):
         """Get the transaction ID.  This differs from the transactions hash as
             given by GetHash.  GetTxid excludes witness data, while GetHash
             includes it. """
-        if self.wit:
-            wit = self.wit
-            self.wit = b''
-            txid = Hash(self.serialize())
-            self.wit = wit
+        if self.wit != CTxWitness():
+            txid = Hash(CTransaction(self.vin, self.vout, self.nLockTime,
+                self.nVersion).serialize())
```

Comment:
Seems pretty dangerous to me to change the meaning of GetHash() on everyone - Bitcoin Core's codebase gives GetHash() the same behavior as before.

How about we remove GetHash() entirely and replace it with GetTxid() and GetWitnessHash()? Better for code to cleanly break than do the wrong thing.